src/eval/mod.rs

```diff
@@ -0,0 +1,101 @@
+use log::{debug, info};
+
+use crate::messages::{Attr, BuildRequest, BuildResponseV1, BuildStatus};
+
+use std::{
+    collections::HashSet,
+    fs::File,
+    io::BufRead,
+    path::{Path, PathBuf},
+    process::{Command, Output},
+};
+
+fn log_command_output(output: Output) {
+    for line in output.stderr.lines() {
+        info!("stderr: {:?}", line)
+    }
+
+    for line in output.stdout.lines() {
+        debug!("stdout: {:?}", line)
+    }
+}
+
+pub fn load_r13y_log(rev: &str) -> Vec<BuildResponseV1> {
+    if let Ok(log_file) = File::open(format!("reproducibility-log-{}.json", rev)) {
+        serde_json::from_reader(log_file).expect("Unable to parse r13y log")
+    } else {
+        Vec::new()
+    }
+}
+
+pub fn eval(instruction: BuildRequest) -> HashSet<PathBuf> {
```

Comment:
This is doing too much, because `eval` doesn't only eval, it also eliminates paths which have been built already. This means that by the time `report` is run, there are no paths returned.

I think this is complicated enough that we may want to have eval return a new data type with the ability to filter out already built derivations, but also the ability to iterate over all the derivations listed.